tests/python_functional/functional_tests/destination_drivers/snmp_destination/001-snmp-destination-v2c-normal-working-compared-cisco-ios-traps-default-values.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+#############################################################################
+# Copyright (c) 2020 One Identity
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+import os
+import time
+
+import pytest
+
+from src.helpers.snmptrapd.snmptrapd import SNMPtrapd
+
+
+CISCO_TIMETICKS = '97881'
+CISCO_FACILITY = "SYS"
+CISCO_SEVERITY = "6"
+CISCO_MESSAGE_TYPE = "CONFIG_I"
+CISCO_MESSAGE = "Configured from console by vty1 (10.30.0.32)"
+
+input_log = "<38>Feb 11 21:27:22 testhost testprogram[9999]: test message\n"
+expected_log = 'iso.3.6.1.4.1.9.9.41.1.2.3.1.2.55 = STRING: "' + CISCO_FACILITY + '"	iso.3.6.1.4.1.9.9.41.1.2.3.1.3.55 = INTEGER: ' + CISCO_SEVERITY + '	iso.3.6.1.4.1.9.9.41.1.2.3.1.4.55 = STRING: "' + CISCO_MESSAGE_TYPE + '"	iso.3.6.1.4.1.9.9.41.1.2.3.1.5.55 = STRING: "' + CISCO_MESSAGE + '"	iso.3.6.1.4.1.9.9.41.1.2.3.1.6.55 = Timeticks: (' + CISCO_TIMETICKS + ')'
+
+
+@pytest.mark.skipif(os.system("which snmptrapd") != 0, reason="snmptrapd can't found on system")
```

Comment:
Instead of this, I would prefer installing snmptrapd into the test docker images.

Reply:
And what if someone wants to run tests in a different  environment (and don't want to install snmptrapd...)?

Reply:
If there is a manual exclude mechanism (blacklist style), I am fine with that. I only have problem with the automatic exlude part.

If the snmptrapd binary somehow disappears unintentionally from our test environment, then these tests are not executed silently, and we would never know it.

We followed this philosophy in https://github.com/syslog-ng/syslog-ng/pull/1878#pullrequestreview-95697739 too.

Reply:
In my mind the output of pytest which states that test case is SKIP doesn't mean it is silent.
Of course, I'm ready to remove this skip mechanism but I afraid it will case unnecessary problems for those users whom doesn't have trapd installed on the system and don't care about snmp destination at all.

Reply:
Yes, the console output can tell which tests are skipped. The problem is that the little box in the PR-s do not show it.

Prior this change, it was enough for us to give a quick glance that all tests are green, and know that the regression passed.
After this change, besides checking that all tests are green, we need to go into the console log too, and look for the snmp tests and make sure they executed. As we cannot be sure anymore which tests are executed, and which are not.

And that's what I would like to avoid.

Reply:
I am agreed with @furiel 
It is more important to see if something starts to fail, then to add a convenient running feature to our testcases. 
But in the other hand we should somehow be able to filter this new type of testcases (which has external dependencies). Pytest has a support to mark (label) the testcases. [Control skipping testcases](https://docs.pytest.org/en/latest/example/simple.html#control-skipping-of-tests-according-to-command-line-option)
I would recommend to introduce a new external running parameter "--skip-tests-with-external-deps". We already have such functionality (with --runslow) https://github.com/syslog-ng/syslog-ng/blob/14a905bb839867afe52109191a6cb366b6a65644/tests/python_functional/conftest.py#L43

Reply:
I also recommend to handle calling snmptrapd binary via ProcessExecutor() class with a `try..except` block maybe a new Exception class can be introduced here `class MissingExternalDependency(Exception)`

Reply:
I added the --skip-tests-with-external-deps. This is False by default, so it will try to run the snmp tests as long as you don't specify the command line option.
I tried to add the snmptrapd package to dbld and travis, but I need some help with this. As far as I see for travis we use only @lbudai OBS repo. I guess we have to add the snmptrapd package to the OBS first. Am I correct?